src/renderer/Renderer.cpp

```diff
@@ -199,6 +201,10 @@ CRenderer::SRenderFeedback CRenderer::renderLock(const CSessionLockSurface& surf
         if (**PNOFADEIN)
             bga = 1.0;
 
+        if (g_pHyprlock->m_bFadeStarted && !**PNOFADEOUT) {
+            bga = std::clamp(std::chrono::duration_cast<std::chrono::microseconds>(g_pHyprlock->m_tFadeEnds - std::chrono::system_clock::now()).count() / 1000000.0 - 0.02, 0.0, 1.0);
```

Comment:
too long, maybe 0.5s?

Reply:
I tried 0.5s but there was a small jump at the end, 0.6s feels perfect though

Reply:
Here's how it looks like:
https://github.com/hyprwm/hyprlock/assets/84299080/a706a300-ff1d-4c27-8fbd-32a8b5caf1e7

Reply:
yea better